db/version_edit_handler.cc

```diff
@@ -909,7 +909,10 @@ void DumpManifestHandler::CheckIterationResult(const log::Reader& reader,
       fprintf(stdout, "comparator: %s\n", cfd->user_comparator()->Name());
     }
     assert(cfd->current());
-    fprintf(stdout, "%s \n", cfd->current()->DebugString(hex_).c_str());
+
+    // Print out DebugStrings. Can include non-terminating null characters.
+    fwrite(cfd->current()->DebugString(hex_).c_str(), sizeof(char),
```

Comment:
Nit: better to use `.data()` instead of `.c_str()` when you're not using it as a null-terminated string

(obviously 2 more like this)